pkg/build/controller/factory/factory.go

```diff
@@ -288,6 +332,61 @@ func (lw *buildLW) Watch(resourceVersion string) (watch.Interface, error) {
 	return lw.client.Builds(kapi.NamespaceAll).Watch(labels.Everything(), fields.Everything(), resourceVersion)
 }
 
+// buildDeleteLW is a ListWatcher implementation that watches for builds being deleted
+type buildDeleteLW struct {
+	client ControllerClient
```

Comment:
Embed the `ControllerClient` without specifying var for it, this way you'll be able to save some typing when accessing those fields, eg.: instead of `lw.client.KubeClient.Pods(...` you'll do `lw.KubeClient.Pods(...`. I'm somehow terrified by those long lines :wink: